# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1507926620 25200
#      Fri Oct 13 13:30:20 2017 -0700
# Node ID 8e034cb195f10889a682d7debed280175332fd89
# Parent  25ccc1f8092dbdcc8ae78ad62d44ff89929efd91
configure: set and restore compiler flags when testing for Qt header files

* acinclude.m4 (OCTAVE_CHECK_QT_OPENGL_OK): Set and restore the C++ compiler
state and options for Qt outside of AC_CACHE_CHECK for testing both the headers
and the compilation test.

diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- a/m4/acinclude.m4
+++ b/m4/acinclude.m4
@@ -1330,25 +1330,29 @@ AC_DEFUN([OCTAVE_CHECK_QT], [
   AM_CONDITIONAL([AMCOND_BUILD_QT_GRAPHICS], [test $build_qt_graphics = yes])
   AM_CONDITIONAL([AMCOND_HAVE_QSCINTILLA], [test $use_qscintilla = yes])
   AM_CONDITIONAL([WIN32_TERMINAL], [test $win32_terminal = yes])
 ])
 dnl
 dnl Check whether Qt works with full OpenGL support
 dnl
 AC_DEFUN([OCTAVE_CHECK_QT_OPENGL_OK], [
+  dnl Normally the language and compiler flags would be set and restored
+  dnl inside of the AC_CACHE_CHECK body.  Because we also need to check for
+  dnl Qt header files associated with the compilation test, set and restore
+  dnl these values outside of the AC_CACHE_CHECK for this macro only.
+  AC_LANG_PUSH(C++)
+  ac_octave_save_CPPFLAGS="$CPPFLAGS"
+  ac_octave_save_CXXFLAGS="$CXXFLAGS"
+  CPPFLAGS="$QT_CPPFLAGS $CXXPICFLAG $CPPFLAGS"
+  CXXFLAGS="$CXXPICFLAG $CXXFLAGS"
   AC_CHECK_HEADERS([QOpenGLWidget QGLWidget])
   AC_CACHE_CHECK([whether Qt works with OpenGL and GLU],
     [octave_cv_qt_opengl_ok],
-    [AC_LANG_PUSH(C++)
-     ac_octave_save_CPPFLAGS="$CPPFLAGS"
-     ac_octave_save_CXXFLAGS="$CXXFLAGS"
-     CPPFLAGS="$QT_CPPFLAGS $CXXPICFLAG $CPPFLAGS"
-     CXXFLAGS="$CXXPICFLAG $CXXFLAGS"
-     AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+    [AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
          #if HAVE_WINDOWS_H
          #  include <windows.h>
          #endif
          #if defined (HAVE_GL_GL_H)
          #  include <GL/gl.h>
          #elif defined (HAVE_OPENGL_GL_H)
          #  include <OpenGL/gl.h>
          #endif
@@ -1371,20 +1375,20 @@ AC_DEFUN([OCTAVE_CHECK_QT_OPENGL_OK], [
              : OCTAVE_QT_OPENGL_WIDGET (parent) { }
            ~gl_widget () {}
          };
          ]], [[
          gl_widget widget;
        ]])],
        octave_cv_qt_opengl_ok=yes,
        octave_cv_qt_opengl_ok=no)
-     CPPFLAGS="$ac_octave_save_CPPFLAGS"
-     CXXFLAGS="$ac_octave_save_CXXFLAGS"
-     AC_LANG_POP(C++)
   ])
+  CPPFLAGS="$ac_octave_save_CPPFLAGS"
+  CXXFLAGS="$ac_octave_save_CXXFLAGS"
+  AC_LANG_POP(C++)
   if test $octave_cv_qt_opengl_ok = yes; then
     $1
     :
   else
     $2
     :
   fi
 ])
